cmd/kube-controller-manager/app/core.go

```diff
@@ -481,9 +482,15 @@ func startTTLAfterFinishedController(ctx ControllerContext) (http.Handler, bool,
 		return nil, false, nil
 	}
 	go ttlafterfinished.New(
-		ctx.InformerFactory.Batch().V1().Jobs(),
-		ctx.ClientBuilder.ClientOrDie("ttl-after-finished-controller"),
-	).Run(int(ctx.ComponentConfig.TTLAfterFinishedController.ConcurrentTTLSyncs), ctx.Stop)
+		ctx.InformerFactory,
+		schema.GroupVersionResource{Group: "batch", Version: "v1", Resource: "jobs"},
+		dynamic.NewForConfigOrDie(ctx.ClientBuilder.ConfigOrDie("ttl-jobs-after-finished-controller")),
+		ctx.ClientBuilder.ClientOrDie("ttl-jobs-after-finished-controller")).Run(int(ctx.ComponentConfig.TTLAfterFinishedController.ConcurrentTTLSyncs), ctx.Stop)
+	go ttlafterfinished.New(
+		ctx.InformerFactory,
+		schema.GroupVersionResource{Version: "v1", Resource: "pods"},
+		dynamic.NewForConfigOrDie(ctx.ClientBuilder.ConfigOrDie("ttl-pods-after-finished-controller")),
+		ctx.ClientBuilder.ClientOrDie("ttl-pods-after-finished-controller")).Run(int(ctx.ComponentConfig.TTLAfterFinishedController.ConcurrentTTLSyncs), ctx.Stop)
```

Comment:
We only need to start a single instance of TTL controller that handles both Jobs and Pods.

Reply:
I agree, start one instance of the controller, 2 informers, 1 queue and then switch based on type where the flow would differ. We can go to dynamic informers and unstructured objects checking if we determine we need it for CRDs later.